tommy-bolger/skeleton-app-vue
# Interview Test - WAC Recipe Search 3000

This PR creates the WAC Recipe Search 3000 SPA with the ability to browse, search, and view recipes. All of the data powering the frontend is fetched from the Laravel API backend.

The backend API has full test coverage with the following test files:
- `RecipeControllerIndexTest`
- `RecipeControllerViewTest`
- `RecipesDatasetTest`

Part of setup executes a seeder that generates 50 recipes using data from https://dummyjson.com/docs/recipes.

The original Docker setup has been heavily modified to more closely mimic a production environment and simplify setup for development. If you need to get into the shell for the PHP container you just need to execute `./shell.sh` on the command line.

Follow these steps to initialize the environment:

1. On the command line, run `docker compose build --no-cache`.

2. Edit `/etc/hosts` (or your OS' hosts file) and add these lines:
127.0.0.1       backend.wild-alaskan.test
127.0.0.1       frontend.wild-alaskan.test

3. Execute `docker compose up -d` to start the environment.

4. Run `./install.sh` to fully configure the environment.

5. Navigate to `backend.wild-alaskan.test` and accept the SSL cert. You should then see the Laravel Welcome page.

6. Navigate to `frontend.wild-alaskan.test` and accept the SSL cert. You should then see the WAC Recipe Search 3000 page and be able to interact with it.
